### PR TITLE
PYTHON-4651: Migrate test_client_context.py to async

### PR DESCRIPTION
**Pull request by sleepyStick**

### CODE REVIEW

**Comment by sleepyStick** [comment]
I think the failing tests are all related to [PYTHON-4701](https://jira.mongodb.org/browse/PYTHON-4701)

**Comment by ShaneHarvey** [comment]
Strange the rhel8-pr-assign-reviewer task ran but didn't assign a reviewer.

**Comment by sleepyStick** [comment]
Oh i assumed it was cuz the tests/checks weren't passing? Because it won't let me assign / add reviews either..?

File: test/asynchronous/test_client_context.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+from test.asynchronous import AsyncUnitTest, SkipTest, async_client_context, unittest
+
+_IS_SYNC = False
+
+
+class TestAsyncClientContext(AsyncUnitTest):
+    @classmethod
+    async def _setup_class(cls):
+        pass
```
Comment (ShaneHarvey) [inline test/asynchronous/test_client_context.py:29]: Can we remove _setup_class since it's just a `pass`?

Comment (sleepyStick) [reply test/asynchronous/test_client_context.py:29]: `AsyncUnitTest` assumes that `_setup_class` would be implemented..without this the code would infinitely recurse and eventually error. 

To my knowledge this is the first time that a class inherits from `AsyncUnitTest` that doesn't "need" a `_setup_class`. I don't know if we want to change `AsyncUnitTest` to accommodate this class's use case or not...

Comment (ShaneHarvey) [reply test/asynchronous/test_client_context.py:29]: AsyncUnitTest should `pass` instead of trying to recursively call itself.

**Comment by blink1073** [comment]
> Strange the rhel8-pr-assign-reviewer task ran but didn't assign a reviewer.

That's intentional in draft mode

**Comment by ShaneHarvey** [comment]
But the PR was moved out of draft mode. Shouldn't the reviewer be auto-assigned in that case?

**Comment by blink1073** [comment]
> But the PR was moved out of draft mode. Shouldn't the reviewer be auto-assigned in that case?

Not unless the EVG task is re-run.

**Comment by ShaneHarvey** [comment]
Oh TIL. I thought it worked the other way.

**Comment by blink1073** [comment]
> Oh TIL. I thought it worked the other way.

Not until we can have other GitHub events trigger EVG runs, or we make our own service listener with a webhook.

File: test/asynchronous/test_client_context.py (hunk 1 above)
Comment (sleepyStick) [reply test/asynchronous/test_client_context.py]: That work's too, don't know why i didn't think of that earlier! Just changed :)

**Review by NoahStapp — CHANGES_REQUESTED** [review]
Great work! The async `test_client_context.py` should also be run [here](https://github.com/mongodb/mongo-python-driver/blob/e6b95f65953e01e435004af069a6976473eaf841/.github/workflows/test-python.yml#L212) as part of the `test_sdist` PR hook.

**Review by ShaneHarvey — DISMISSED** [review]

File: .github/workflows/test-python.yml [hunk 2]
```diff
@@ -209,4 +209,4 @@ jobs:
             ls
             which python
             pip install -e ".[test]"
-            PYMONGO_MUST_CONNECT=1 pytest -v test/test_client_context.py
+            PYMONGO_MUST_CONNECT=1 pytest -k client_context
```
Comment (NoahStapp) [inline .github/workflows/test-python.yml:212]:
```suggestion
            PYMONGO_MUST_CONNECT=1 pytest -v -k client_context
```

Comment (NoahStapp) [reply .github/workflows/test-python.yml:212]: For consistency across our test runs.

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .github/workflows/test-python.yml (hunk 2 above)
Comment (ShaneHarvey) [reply .github/workflows/test-python.yml:212]: Is that something we can set in pyproject?

**Review by ShaneHarvey — APPROVED** [review]

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**